apiBase/apiRequest.js

```diff
@@ -0,0 +1,36 @@
+import superagent from 'superagent';
+
+import APIResponse from './APIResponse';
+
+
+export default (apiOptions, method, path, query={}, body={}, type=null) => {
+  const { origin, appName, env, token, defaultHeaders={} } = apiOptions;
+
+  const _method = method.toLowerCase();
+  const _headers = token ?
+    { ...defaultHeaders, Authorization: `Bearer ${token}` } : defaultHeaders;
+  const _path = path.startsWith('/') ? path : `/${path}`
+  const _query = { ...query, app: `${appName}-${env}` };
+  const endpoint = origin + _path;
+
+  const request = superagent(_method, endpoint).set(_headers).query(_query);
+
+  if (type) {
+    request.type(type);
+  }
+
+  if (_method === 'post') {
+    request.send(body);
+  }
+
+  return new Promise((resolve, reject) => {
+    request.end((err, res) => {
+      if (!err) {
+        resolve(new APIResponse(res));
+      } else {
+        reject(err);
```

Comment:
per convo with Ryan, we should reject `ResponseError` here to normalize superagent's error responses.